### DISCUSSION TITLE
About ui textures.

### DISCUSSION THREAD

**Discussion by ghost** [FaithBeam/Sims-1-Complete-Collection-Widescreen-Patcher#46] [Ideas]
Hi, first of all I want to say that I admire your mod, which easily helps the game to become full screen and provides at once DdRawCompat, with which the game runs faster. In the last update you added a texture fix that I don't quite understand. Yes I saw that the icon textures are different from the panel textures, but I used PanekBack from here and it looked fine to me: **https://www.reddit.com/r/thesims/comments/6snibn/the_sims_1_widescreen_fix_1080p/**

**Comment by ghost** [comment]
Also, I saw the ui-mod: **https://classicsims.weebly.com/hd-ui-mod.html**. It looks fun, but I haven't been able to contact the author. The images on the site are not presented in full hd, but they can technically be stretched with special services like waifu2x and the like. Do you think it would be possible to try to adapt this work for the game?

**Reply by FaithBeam** [reply]
I'm not sure how I'd go about this.

1. I don't know where the textures for objects are stored but I bet I could find them and make a library to extract them
2. I'm not a game dev or 3d modeler so I don't know how texture sizes should relate to the model they're attached to
3. I wouldn't be distributing the textures as a download but would include the ability to extract them and upscale them on your own computer. Reason for that is I don't want to host copyrighted material which could be used to take this repo down however small that chance may be. This means doing this patch would take a long long time, imagine someone's laptop from 2010 running waifu2x on hundreds of files with no gpu acceleration lol

I'm not against it, there are just multiple hurdles that need to be figured out.

**Reply by ghost** [reply]
I apologize for the mistakes, as English is not my first language.
The textures are stored in uigraphics.far, which can be edited by FarX or similar programs, and you can use dragon unpacker to view them. To make the textures for the neighborhood map work, you actually need to translate the custom images to bmp and put a purple square in the center and it seems to work (Replacing the largeback file, as I understand it). It's ok if you don't want to do it, thanks for the patch

**Comment by FaithBeam** [comment]
> Hi, first of all I want to say that I admire your mod, which easily helps the game to become full screen and provides at once DdRawCompat, with which the game runs faster. In the last update you added a texture fix that I don't quite understand. Yes I saw that the icon textures are different from the panel textures, but I used PanekBack from here and it looked fine to me: **https://www.reddit.com/r/thesims/comments/6snibn/the_sims_1_widescreen_fix_1080p/**

Ha, you're absolutely right. The difference between what I did vs what they did in your link is I do a blind resize on the panelback. What they did is extend the dark blue section to 1920 which preserves the original background color of the icons. It's also more vanilla looking than mine. View the comparison here: https://imgsli.com/MTAyNjY0

Thank you for pointing this out. I don't know how I'll make this change with Magick.net but I'll look into it.

**Reply by FaithBeam** [reply]
@PieStars I made this change in [Release 2.3.0](https://github.com/FaithBeam/Sims-1-Complete-Collection-Widescreen-Patcher/releases/tag/2.3.0). The panelback.bmp is properly resized like in the link you provided so there is no need to fix the transparency of icons. Thank you very much! :)

**Reply by 130N1D** [reply]
> @PieStars I made this change in [Release 2.3.0](https://github.com/FaithBeam/Sims-1-Complete-Collection-Widescreen-Patcher/releases/tag/2.3.0). The panelback.bmp is properly resized like in the link you provided so there is no need to fix the transparency of icons. Thank you very much! :)

Yay! Thank for the update, now everything is fine, and there are no graphical bugs when moving the camera in window mode, as well as problems with icons in buy/build modes, but now there is another problem, I play in 2560x1440 resolution and your soft does not correctly generate texture Panelback.bmp, there is a small artifact in the center, I successfully corrected my bmp in photoshop. 

https://imgur.com/a/3MY6XKw

**Reply by 130N1D** [reply]
Yep, just checked, every resolution above 1920x1080 has this artifact

**Reply by FaithBeam** [reply]
@130N1D Thank you for finding that out. I should've tested higher resolutions than 1920x1080. The reason the same bug affects resolutions higher than that is because it internally loads panelback.bmp into memory twice. It resizes one of them to the width you want, so, if you chose 2560 as the width, it resizes that image to 2560x100. Then it lays the other unresized image on top of the one the program resized earlier. The resolution of the unedited panelback.bmp is 804x100. It so happens that when the first image is resized greater than 1920, the white space starts to peak out from the image underneath.

This problem is now more difficult because I think it will involve splitting panelback.bmp into 3 separate images in memory, and resizing the middle section which will (finally? :D) solve this issue. I'll get to work on it.

**Reply by FaithBeam** [reply]
@130N1D I released [2.3.1](https://github.com/FaithBeam/Sims-1-Complete-Collection-Widescreen-Patcher/releases/tag/2.3.1) which fixes this issue. This is 1440p: https://i.imgur.com/AkFqAR4.jpeg
